chain-signatures/node/src/protocol/signature.rs

```diff
@@ -227,18 +211,35 @@ pub struct GenerationRequest {
     pub proposer: Participant,
     pub request: ContractSignRequest,
     pub epsilon: Scalar,
-    pub receipt_id: CryptoHash,
+    pub receipt_id: [u8; 32],
     pub entropy: [u8; 32],
     pub sign_request_timestamp: Instant,
 }
 
+#[derive(Debug, Clone, Eq, Hash, PartialEq)]
+pub struct SignRequestIdentifier {
+    pub receipt_id: [u8; 32],
+    pub epsilon: Vec<u8>,
+    pub payload: Vec<u8>,
+}
+
+impl SignRequestIdentifier {
+    pub fn new(receipt_id: [u8; 32], epsilon: Scalar, payload: Scalar) -> Self {
+        Self {
+            receipt_id,
+            epsilon: borsh::to_vec(&SerializableScalar { scalar: epsilon }).unwrap(),
+            payload: borsh::to_vec(&SerializableScalar { scalar: payload }).unwrap(),
+        }
+    }
+}
```

Comment:
why are we making epsilon and the payload vectors? Is it for hashing? If so, just implement hashing for `SerializableScalar`

Reply:
This will come in a refactoring PR